File: 189. Rotate Array/189.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        
        var tmp = new int[nums.Length];
        int l = nums.Length;
        k %= l;
        Array.Copy(nums, l-k, tmp, 0, k );
        Array.Copy(nums, 0, nums, k, l-k);
        Array.Copy(tmp, 0, nums, 0, k);

    }
}

File: 189. Rotate Array/189.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        left = 0
        right = len(nums) - 1
        for i in range(k):
            nums.insert(0, nums[-1])
            nums.pop(-1)